components/src/core/DropdownToolbar/DropdownToolbar.tsx

```diff
@@ -0,0 +1,228 @@
+            lineHeight: 1,
+            textOverflow: 'ellipsis',
+            overflow: 'auto',
+        },
+        subtitleContent: {
+            display: 'flex',
+            flexDirection: 'row',
+            maxWidth: 'fit-content',
+        },
+        cursorPointer: {
+            cursor: 'pointer',
+        },
+        subtitle: {
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+        },
+        dropdownArrow: {
+            marginLeft: theme.spacing(0.5),
+        },
+        rotateDropdownArrow: {
+            transform: 'rotate(180deg)',
+        },
+    })
+);
+
+export type DropdownToolbarClasses = {
+    root?: string;
+    navigation?: string;
+    textContent?: string;
+    title?: string;
+    subtitleContent?: string;
+    subtitle?: string;
+    dropdownArrow?: string;
+    menuItem?: string;
+};
+
+export const DropdownToolbar: React.FC<DropdownToolbarProps> = (props) => {
+    const {
+        title,
+        subtitle,
+        menuGroups,
+        navigationIcon,
+        menu,
+        MenuProps,
+        onClose,
+        onOpen,
+        classes = {},
+        ...toolbarProps
+    } = props;
+    const [anchorEl, setAnchorEl] = useState(null);
+    const anchor = useRef(null);
+    const theme = useTheme();
+    const defaultClasses = useStyles(theme);
+    const getNavigationIcon = useCallback(() => {
+        if (navigationIcon) {
+            return <div className={clsx(defaultClasses.navigation, classes.navigation)}>{navigationIcon}</div>;
+        }
+    }, [navigationIcon]);
+
+    const closeMenu = useCallback(() => {
+        if (onClose) {
+            onClose();
+        }
+        setAnchorEl(null);
+    }, [onClose]);
+
+    const openMenu = useCallback(
+        (event: MouseEvent) => {
+            if (onOpen) {
+                onOpen();
+            }
+            setAnchorEl(event);
+        },
+        [onOpen]
+    );
+
+    const getMenu = useCallback(() => {
+        if (menu && Boolean(anchorEl)) {
+            return React.cloneElement(menu, {
+                anchorEl: anchorEl,
+                open: Boolean(anchorEl),
+                onClose: closeMenu,
+                ...menu.props,
+            });
+        }
+        if (menuGroups && Boolean(anchorEl)) {
+            return (
+                <Menu
+                    elevation={0}
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+                    transformOrigin={{ vertical: 'top', horizontal: 'left' }}
+                    {...MenuProps}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => closeMenu()}
+                    MenuListProps={{ style: { padding: 0 } }}
+                    PaperProps={{
+                        style: {
+                            maxHeight: MENU_ITEM_HEIGHT * 4.5,
+                        },
+                    }}
```

Comment:
Skip the paper props...if they wan't to set a max height we'll let them do it themselves